### ISSUE TITLE
Cursor jumps back to first index during autocomplete

### ISSUE THREAD

**Issue by bastiankistner** [getinsomnia/insomnia#3437]
**Describe the bug**
The cursor almost always jumps to the first index after ~ 1 second when I am navigating through the autocomplete dialog / overlay.

**To Reproduce**
1. Create any GraphQL request
2. write your query against a schema that you have previously fetched
3. press `control + space` to show autocomplete dialog and quickly move with cursor keys to an item that is not the first in list
4. see the cursor jump back to the first index

**Expected behavior**
Cursor does not jump to first index.

**Screenshots**
![2021-06-02 08 54 26](https://user-images.githubusercontent.com/1783541/120437981-11194b80-c381-11eb-9028-2d582484839c.gif)

**Desktop (please complete the following information):**
 - OS: macOS 11.4
 - Installation Method: https://insomnia.rest/download

```
Version: Insomnia 2021.3.0
Release date: 4/28/2021
OS: Darwin x64 20.5.0
Electron: 9.3.5
Node: 12.14.1
V8: 8.3.110.13-electron.0
Architecture: x64
node-libcurl: libcurl/7.73.0 OpenSSL/1.1.1j zlib/1.2.11 brotli/1.0.9 zstd/1.4.9 libidn2/2.1.1 libssh2/1.9.0 nghttp2/1.41.0
```

**Comment by bastiankistner** [comment]
I wonder why no one else seems to have stumbled upon this. I've been having this issue now for at least half a year (maybe even longer). It was so annoying that I temporarily switched to postman.

**Comment by MrSnix** [comment]
Hi, 👋🏻  thanks for reporting.
I tried to give it a shoot, but I couldn't reproduce it, but I want to dig into it further.
Seems a tricky one pending for a long time, maybe I may fix it.

Duplicate of #2585 and #1015

**EDIT**: 
**I was able to reproduce it** but Woah ... on 15 tries only 1 time it happened. Sounds really cool this bug.

**EDIT - 2**: 
It's interesting to see the consistency of this differing system by system. 
On some computer, it happens a lot of the time, on others quite always, on mine quite never.

**EDIT - 3**:
After so many hours of debugging I found out why this reset happens, now I have to figure out what is the correct way to fix this thing.
